clients/venice-push-job/src/main/java/com/linkedin/venice/spark/datawriter/jobs/AbstractDataWriterSparkJob.java

```diff
@@ -421,62 +458,62 @@ private void logAccumulatorValue(AccumulatorV2<?, ?> accumulator) {
     LOGGER.info("  {}: {}", accumulator.name().get(), accumulator.value());
   }
 
-  private void validateDataFrame(Dataset<Row> dataFrameForDataWriterJob) {
+  @VisibleForTesting
+  void validateDataFrame(Dataset<Row> dataFrameForDataWriterJob) {
     if (dataFrameForDataWriterJob == null) {
       throw new VeniceInvalidInputException("The input data frame cannot be null");
     }
 
     StructType dataSchema = dataFrameForDataWriterJob.schema();
     StructField[] fields = dataSchema.fields();
 
-    if (fields.length < 2) {
+    if (fields.length < 3) {
       String errorMessage =
           String.format("The provided input data does not have enough fields. Provided schema: %s.", dataSchema);
       throw new VeniceInvalidInputException(errorMessage);
     }
 
-    int keyFieldIndex = SparkScalaUtils.getFieldIndex(dataSchema, KEY_COLUMN_NAME);
+    validateDataFrameFieldAndTypes(fields, dataSchema, KEY_COLUMN_NAME, Collections.singleton(DataTypes.BinaryType));
+    validateDataFrameFieldAndTypes(fields, dataSchema, VALUE_COLUMN_NAME, Collections.singleton(DataTypes.BinaryType));
 
-    if (keyFieldIndex == -1) {
-      String errorMessage = String.format(
-          "The provided input data frame does not have a %s field. Provided schema: %s.",
-          KEY_COLUMN_NAME,
-          dataSchema);
-      throw new VeniceInvalidInputException(errorMessage);
-    }
+    validateDataFrameFieldAndTypes(
+        fields,
+        dataSchema,
+        RMD_COLUMN_NAME,
+        ImmutableList.of(DataTypes.BinaryType, DataTypes.LongType));
```

Comment:
This has now changed to `byte[]` only, right? We should revert the list, and simplify the function to always expect one type for now